File: frontend/lib/models/received_friend_request_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'received_friend_request_dto.g.dart';

@JsonSerializable()
class ReceivedFriendRequestDTO {
  final String requestId; // UUID'ler String olarak temsil edilir
  final String senderPhotoUrl; // Gönderenin profil fotoğrafının URL'si
  final String senderUsername; // Gönderenin kullanıcı adı
  final String senderFullName; // Gönderenin tam adı
  @JsonKey(name: 'sentAt')
  final String sentAt; // İsteğin gönderildiği tarih/saat (formatlanmış)
  final int popularityScore; // Gönderenin popülerlik skoru

  ReceivedFriendRequestDTO({
    required this.requestId,
    required this.senderPhotoUrl,
    required this.senderUsername,
    required this.senderFullName,
    required this.sentAt,
    required this.popularityScore,
  });

  factory ReceivedFriendRequestDTO.fromJson(Map<String, dynamic> json) => _$ReceivedFriendRequestDTOFromJson(json);
  Map<String, dynamic> toJson() => _$ReceivedFriendRequestDTOToJson(this);
} 

File: backend/src/main/java/bingol/campus/friendRequest/core/response/ReceivedFriendRequestDTO.java
package bingol.campus.friendRequest.core.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ReceivedFriendRequestDTO {
    private UUID requestId;           // Arkadaşlık isteği ID'si
    private String senderPhotoUrl;    // Gönderenin profil fotoğrafının URL'si
    private String senderUsername;    // Gönderenin kullanıcı adı
    private String senderFullName;    // Gönderenin tam adı
    private String sentAt;            // İsteğin gönderildiği tarih/saat (formatlanmış)
    private long popularityScore;
}
